crates/iceberg/src/transform/temporal.rs

```diff
@@ -51,12 +61,48 @@ impl TransformFunction for Year {
                 .unary(|v| v - UNIX_EPOCH_YEAR),
         ))
     }
+
+    fn transform_literal(
+        &self,
+        input: &crate::spec::Literal,
+    ) -> Result<Option<crate::spec::Literal>> {
+        match input {
+            Literal::Primitive(PrimitiveLiteral::Date(v)) => Ok(Some(Literal::Primitive(
+                PrimitiveLiteral::Int(Date32Type::to_naive_date(*v).year() - UNIX_EPOCH_YEAR),
+            ))),
+            Literal::Primitive(PrimitiveLiteral::Timestamp(v)) => Ok(Some(Literal::Primitive(
+                PrimitiveLiteral::Int(Self::timestamp_to_year(*v)),
+            ))),
+            Literal::Primitive(PrimitiveLiteral::TimestampTZ(v)) => Ok(Some(Literal::Primitive(
+                PrimitiveLiteral::Int(Self::timestamp_to_year(*v)),
+            ))),
+            _ => unreachable!("Should not call internally for unsupported literal type"),
+        }
+    }
 }
 
 /// Extract a date or timestamp month, as months from 1970-01-01
 #[derive(Debug)]
 pub struct Month;
 
+impl Month {
+    #[inline]
+    fn timestamp_to_month(timestamp: i64) -> i32 {
```

Comment:
This is incorrect, we should the month is accumulated month, see https://github.com/apache/iceberg/blob/9d42eb44a32ff4a1f86e8427fe023bde1e9652f7/api/src/test/java/org/apache/iceberg/transforms/TestDates.java#L56